core/serialization/deserializer/montage-reviver.js

```diff
@@ -419,7 +415,7 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
                 }
                 return element;
             } else {
-                return Promise.reject(new Error("Element with id '" + elementId + "' was not found."));
+                throw new Error("Element with id '" + elementId + "' was not found.");
```

Comment:
shouldn't there be test if not sync then Promise.reject(error) as before?

Reply:
It's not strictly necessary because the montage-deserializer does a try/catch over context API calls to catch synchronous errors as well as rejected promises. Since nothing but the deserializer uses context APIs, why create an extra promise when we could just throw a regular error?